python/triton/compiler.py

```diff
@@ -394,7 +394,15 @@ def visit_If(self, node):
                             if then_defs[then_name].type == else_defs[else_name].type:
                                 names.append(then_name)
                                 ret_types.append(then_defs[then_name].type)
-
+                
+                # defined in else block but not in then block
```

Comment:
what will happen if some value is defined in then block but not in else block?

Reply:
has been handled in original code